substrate/frame/contracts/Cargo.toml

```diff
@@ -48,20 +48,34 @@ sp-io = { path = "../../primitives/io", default-features = false}
 sp-runtime = { path = "../../primitives/runtime", default-features = false}
 sp-std = { path = "../../primitives/std", default-features = false}
 
+xcm = { package = "staging-xcm", path = "../../../polkadot/xcm", default-features = false}
+xcm-executor = { package = "staging-xcm-executor", path = "../../../polkadot/xcm/xcm-executor", default-features = false}
+pallet-xcm = { path = "../../../polkadot/xcm/pallet-xcm", default-features = false}
+
 [dev-dependencies]
 array-bytes = "6.1"
 assert_matches = "1"
 env_logger = "0.9"
 pretty_assertions = "1"
 wat = "1"
 
+# Polkadot Dependencies
+xcm-simulator = {path = "../../../polkadot/xcm/xcm-simulator"}
+xcm-builder = {package = "staging-xcm-builder", path = "../../../polkadot/xcm/xcm-builder"}
+polkadot-runtime-parachains = {path = "../../../polkadot/runtime/parachains"}
+polkadot-parachain-primitives = { path = "../../../polkadot/parachain"  }
+polkadot-primitives = { path = "../../../polkadot/primitives"  }
+
```

Comment:
I think this is wrong: see #1318 

Is it possibly to remove any dependency outside Substrate here? If not, why?

Unless it is now allowed to commingle any dependency in the SDK, that would erode the notion of any separation of the three previously (intentionally) separate repos. I hope that isn't the case?

Reply:
These dependencies are used to test the pallet in a mocked network environment.
Not sure why this is an issue here since this is only used for testing purpose.

Reply:
Not my call, but testing relying on things downstream for Substrate seems wrong to me. I don't see other libraries calling their consumers to test their features... maybe this is actually idiomatic and OK? :shrug:

Reply:
I disagree with the **opinion** in your issue that this graph should be enforced. Big point of the monorepo was that which crates are in polkadot vs. cumulus vs. substrate is an arbitrary and ultimately useless distinction. Fact of the matter is that we need those crates.

Reply:
@NukeManDan  all these deps apart from `xcm` and `xcm-builder` are now moved out to the testing crate `pallet-contracts-mock-network`.

`xcm` and `xcm-builder` should eventually be moved out of `polkadot` and into it's own repo, and hopefully be adopted by other substrate pallets.

Reply:
Thanks for the note @pgherveou - that is good news that things are moving away from any `./substrate/.....` paths depending on `../../../polkadot/.....` ones. There are a lot remaining in here so my opinion stands the same: #1318 . But as already pointed out, it is just an opinion. Getting this in if the team thinks it should not wait for the "right" thing should not depend on a approval from me (if it does, please remove my review to allow this to merge)